zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -0,0 +1,511 @@
+                accs[1]
+                    .address(caller.address)
+                    .code(caller.code)
+                    .nonce(caller.nonce)
+                    .balance(caller.balance);
+                accs[2]
+                    .address(callee.address)
+                    .code(callee.code)
+                    .nonce(callee.nonce)
+                    .balance(callee.balance);
+            },
+            |mut txs, accs| {
+                txs[0]
+                    .from(accs[0].address)
+                    .to(accs[1].address)
+                    .gas(tx_gas.into());
+            },
+            |block, _tx| block.number(0xcafeu64),
+        )
+        .unwrap()
+        .into();
+        let block_data = bus_mapping::mock::BlockData::new_from_geth_data(block);
+        let mut builder = block_data.new_circuit_input_builder();
+        builder
+            .handle_block(&block_data.eth_block, &block_data.geth_traces)
+            .unwrap();
+        let block = block_convert(&builder.block, &builder.code_db);
+        assert_eq!(run_test_circuit(block), Ok(()));
+    }
+
+    #[test]
+    fn call_with_oog_root() {
+        let stacks = vec![
+            // With gas and memory expansion
+            Stack {
+                gas: 100,
+                cd_offset: 64,
+                cd_length: 320,
+                rd_offset: 0,
+                rd_length: 32,
+                ..Default::default()
+            },
+        ];
+
+        let bytecode = bytecode! {
+            PUSH32(Word::from(0))
+            PUSH32(Word::from(0))
+            STOP
+        };
+        let callees = vec![callee(bytecode)];
+        for (stack, callee) in stacks.into_iter().cartesian_product(callees.into_iter()) {
+            test_oog(caller(stack, true), callee, true);
+        }
+    }
+
+    #[test]
+    fn call_with_oog_internal() {
+        let stacks = vec![
+            // first call stack
+            Stack {
+                gas: 100,
+                cd_offset: 64,
+                cd_length: 320,
+                rd_offset: 0,
+                rd_length: 32,
+                ..Default::default()
+            },
+            // second call stack
+            Stack {
+                gas: 21,
+                cd_offset: 64,
+                cd_length: 320,
+                rd_offset: 0,
+                rd_length: 32,
+                ..Default::default()
+            },
+        ];
+
+        let stack = stacks[1];
+        let bytecode = bytecode! {
+            PUSH32(Word::from(stack.rd_length))
+            PUSH32(Word::from(stack.rd_offset))
+            PUSH32(Word::from(stack.cd_length))
+            PUSH32(Word::from(stack.cd_offset))
+            PUSH32(stack.value)
+            PUSH32(Address::repeat_byte(0xfe).to_word())
+            PUSH32(Word::from(stack.gas))
+            CALL // make this call out of gas
+            PUSH32(Word::from(0))
+            PUSH32(Word::from(0))
+        };
```

Comment:
These are never used and unnecessary

Reply:
these are used for callee bytecode in L508

Reply:
They will never be executed, since the call from `caller` is out of gas.

Reply:
seems not,  I tried remove them and set callee bytecode just `STOP`, this condition will not met oog gas state.

Reply:
I see, just checked the steps and saw it now. I'd still suggest to simply these test cases to make it as simpler as possible (e.g. make call of `caller` out of gas, so `callee` doesn't need to hold bytecode), that should make review much easier.